opentelemetry-sdk/src/opentelemetry/sdk/trace/export/__init__.py

```diff
@@ -72,11 +73,15 @@ def on_start(self, span: Span) -> None:
         pass
 
     def on_end(self, span: Span) -> None:
+        suppress_instrumentation = Context.suppress_instrumentation
         try:
+            Context.suppress_instrumentation = True
             self.span_exporter.export((span,))
         # pylint: disable=broad-except
         except Exception as exc:
             logger.warning("Exception while exporting data: %s", exc)
+        finally:
```

Comment:
Minor: I was a bit struggling whether to put this in `finally` statement or not.
```python
suppress_instrumentation = Context.suppress_instrumentation
try:
    Context.suppress_instrumentation = True
    self.span_exporter.export((span,))
# pylint: disable=broad-except
except Exception as exc:
    logger.warning("Exception while exporting data: %s", exc)
Context.suppress_instrumentation = suppress_instrumentation
```
This looks nicer (symmetric), however if there is uncaught exception, or later someone added a return in the `try` statement, the context could get wrong.

Reply:
And introducing a context manager seems to be an overkill here.

Reply:
I agree with the `finally` here, but also think a context manager in `context` would make sense.

```python
with Context.set('suppress_instrumentation', True):
    # do stuff that shouldn't be instrumented
```

This would also let you use the set/reset token, which you can't access here since `Context` doesn't expose the `ContextVar`s it uses under the hood.